src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Cache/CacheAspectThreadManagementIntegrationTests.cs

```diff
@@ -0,0 +1,589 @@
+#nullable enable
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Cache;
+using GovUk.Education.ExploreEducationStatistics.Common.Cache.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Fixtures;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Tests.Cache;
+
+[Collection(CacheTestFixture.CollectionName)]
+public class CacheAspectThreadManagementIntegrationTests : 
+    IClassFixture<TestApplicationFactory<TestStartup>>,
+    IClassFixture<CacheTestFixture>,
+    IDisposable
+{
+    // An artificial delay that we will add to dummy async sections of our test setup, including getting
+    // cached items setting cached items and time spent in Controller methods. 
+    public const int AsyncOperationDelayMillis = 250;
+    
+    // A reasonable amount of time for code to reach the next step of the HTTP request processing workflow if
+    // no async delaying action appears between the two steps. 
+    private const int NoAsyncDelayTimeMillis = 50;
+    
+    // An error tolerance in the time it takes for the code after a Task.Delay() to  continue running. Task.Delay() is
+    // not 100% accurate in terms of delay and so it is possible for the underlying mechanisms to wake early by a few
+    // milliseconds.
+    private const int TaskDelayErrorToleranceMillis = 15;
+
+    // A list of events that we will capture in order to verify that the flow of code is showing what we expect
+    // to see, especially in terms of allowing threads to be freed up and not be blocked. 
+
```

Comment:
Think the intention is to comment a list here but there's no list 🙂

Reply:
Ahh yup, so this was when EventList was a static field here, but I left the comment behind!  I have moved the comment to where the new List<Event> variables are (where they're actually useful to have for the purposes of understanding the relevant tests anyway).